course_info/tests/jasmine/controllers/PeopleControllerSpec.js

```diff
@@ -74,61 +56,632 @@ describe('Unit testing PeopleController', function() {
+
+            // now resolve the memberDeferred
+            memberDeferred.resolve([]);
+            scope.$digest();
+
+            // and affirm it was called, so we know it's not just a broken test
+            expect(scope.addNewMember).toHaveBeenCalledTimes(2);
+        });
+
         it('won\'t bother waiting for profile lookups or POSTing new members ' +
-            'if course membership lookup fails');
+                'if course membership lookup fails', function() {
+            // promise returned from lookupCourseMembers
+            var memberPromise = $q.reject(new Error());
+            // promises returned from lookupPeople
+            var peoplePromises = [$q.defer().promise, $q.defer().promise];
+
+            // set up spies on methods addPeopleToCourse calls out to
+            spyOn(scope, 'lookupCourseMembers').and.returnValue(memberPromise);
+            spyOn(scope, 'lookupPeople').and.returnValue(peoplePromises);
+
+            // this is the one we're making sure doesn't get called
+            spyOn(scope, 'addNewMember');
+
+            // and this is the one that should
+            spyOn(scope, 'showAddNewMemberResults');
+
+            // run it
+            scope.addPeopleToCourse(searchTermList);
+            scope.$digest();
+
+            // make sure the POST wasn't done
+            expect(scope.addNewMember).not.toHaveBeenCalled();
+
+            // and that the results were updated, despite the profile lookups
+            // being unresolved.
+            expect(scope.showAddNewMemberResults).toHaveBeenCalled();
+        });
     });
+
     describe('addNewMember', function() {
-        it('won\'t add person if they already have a course enrollment');
-        it('won\'t add if they could not be found via /people lookup');
-        it('won\'t add if multiple profiles returned by /people lookup');
-        it('adds person if not in course and single profile available');
+
+        var personResult = [];
+        var members = {};
+
+        beforeEach(function() {
+            controller = $controller('PeopleController', {$scope: scope });
+            //var members = {};
+            members["12345678"] = [{
+                        active: 1,
+                        email_address: "test45@mcelroy.org",
+                        name_first: "Test",
+                        name_last: "User",
+                        prime_role_indicator: "Y",
+                        role_type_cd: "STUDENT",
+                        univ_id: "12345678"
+                    }];
+            members["43215678"] = [{
+                        active: 1,
+                        email_address: "test46@mcelroy.org",
+                        name_first: "Peter",
+                        name_last: "Smith",
+                        prime_role_indicator: "Y",
+                        role_type_cd: "STUDENT",
+                        univ_id: "43215678"
+                    }];
+
+            personResult.push(
```

Comment:
Is the intention to continue to push things onto the `personResult`, such that by the third test `personResult.length == 6`? If not, I think this can also be moved outside of the `beforeEach()`, if it's not being altered somehow in the tests.